Frameworks/MQTTnet.NetFramework/MqttServerFactory.cs

```diff
@@ -8,8 +9,9 @@ public class MqttServerFactory
         public MqttServer CreateMqttServer(MqttServerOptions options)
         {
             if (options == null) throw new ArgumentNullException(nameof(options));
-
-            return new MqttServer(options, new MqttServerAdapter());
+            
+            // The cast to IMqttServerAdapter is required... stupidly...
+            return new MqttServer(options, options.UseSSL ? (IMqttServerAdapter) new MqttSslServerAdapter() : new MqttServerAdapter());
```

Comment:
I like this idea, though it would take a bit of work to implement.  Naturally, the encrypted endpoint should be prioritized.  The server code would need to be refactored to permit multiple adapters, but that shouldn't be terribly difficult.